source/common/router/router.cc

```diff
@@ -1602,7 +1602,7 @@ void Filter::onUpstreamComplete(UpstreamRequest& upstream_request) {
     }
   }
 
-  upstream_request.removeFromList(upstream_requests_);
+  callbacks_->dispatcher().deferredDelete(upstream_request.removeFromList(upstream_requests_));
```

Comment:
Is this change to defer delete the request intentional? Seems unrelated to the name change

Reply:
nope sorry, bad PR split thanks for catching